### DISCUSSION TITLE
📢 How Are You Using react-file-manager? Share Your Thoughts!

### DISCUSSION THREAD

**Discussion by Saifullah-dev** [Saifullah-dev/react-file-manager#110] [General]
Hello everyone! 👋

First off, thank you for checking out **react-file-manager**, the open-source file manager component designed to simplify file management integration in your applications. Your interest and support mean the world to me!

I’d love to hear from you—the developers and users who make this project worthwhile.

## 💡 How Is react-file-manager Helping You?

- Are you using it in a personal project, startup, or enterprise-level application?

- Which features do you find most useful?

- What problems has it helped you solve?

## ✨ Room for Improvement

While I’m actively working on ongoing development and maintenance, your feedback can take this project to the next level.

- What’s missing?

- Are there any pain points or bugs you’ve encountered?

- Do you have a new feature idea that would make this even more powerful for your use case?

## 🚀 Let’s Build This Together

The goal is to make react-file-manager the best tool for file management out there, and that can only happen with your input and suggestions. I’d love for you to:

- Share your thoughts here!

- Open an issue if you have specific feedback or feature requests.

- Spread the word to others who might find this useful.

## 🙌 Thank You!

Your feedback, ideas, and contributions will help shape the future of react-file-manager. Let’s create something awesome together!

Looking forward to hearing your thoughts!

Cheers,
Saifullah Zubair
[React File Manager](https://github.com/Saifullah-dev/react-file-manager)

**Comment by novotny** [comment]
Hi @Saifullah-dev,

Thank you for your work on this package! It looks great and suits my needs well.

I just have one small request. My use case involves a virtual folder structure, which makes it impossible to display a preview of images using the default approach, as "filePreviewPath + image[i].path" doesn’t match my image URLs. Instead, I receive the image preview URL directly from the API.

Would it be possible to update your package to support this scenario? For example, by extending ApiFileItem with a thumbnailUrl property or by providing a custom render method for files so that I can specify how the file preview should be displayed?

Thank you for considering this!

Best regards,

Ondra

**Reply by Saifullah-dev** [reply]
@novotny Yes, it is indeed possible to support this scenario by providing a custom render method for file preview. Thank you for pointing it out. I'm going to add it & will notify you upon version release :)

I'm glad you're finding this package useful 🙌

**Reply by Saifullah-dev** [reply]
@novotny [Custom File Preview](https://github.com/Saifullah-dev/react-file-manager?tab=readme-ov-file#custom-file-preview) is now available in the version [v1.14.0](https://www.npmjs.com/package/@cubone/react-file-manager/v/1.14.0).

## Custom File Preview

The `FileManager` component allows you to provide a custom file preview by passing the
`filePreviewComponent` prop. This is an optional callback function that receives the selected file
as an argument and must return a valid React node, JSX element, or HTML.

### Usage Example

```jsx
const CustomImagePreviewer = ({ file }) => {
  return <img src={`${file.path}`} alt={file.name} />;
};

<FileManager
  // Other props...
  filePreviewComponent={(file) => <CustomImagePreviewer file={file} />}
/>;
```

**Reply by novotny** [reply]
@Saifullah-dev Nice! Thank you for very quick answer and the new feature! :)

**Comment by Pradeepdbe** [comment]
Need event for copy function.
as i observed nothing happening on click of copy.

**Reply by Saifullah-dev** [reply]
@Pradeepdbe Thanks for the suggestion. `onItemCopy` prop will be available in the next version release.

**Reply by Saifullah-dev** [reply]
:tada: @Pradeepdbe This Feature is included in version 1.17.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@cubone/react-file-manager/v/1.17.0)
- [GitHub release](https://github.com/Saifullah-dev/react-file-manager/releases/tag/v1.17.0)
